test/ExternalGenericMetadataBuilder/ExternalGenericMetadataBuilder.swift

```diff
@@ -0,0 +1,76 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-build-swift -I %swift-lib-dir -I %swift_src_root/include/swift/ExternalGenericMetadataBuilder -L%swift-lib-dir -lswiftGenericMetadataBuilder %s -o %t/ExternalMetadataBuilderTest
+// RUN: %target-codesign %t/ExternalMetadataBuilderTest
+// RUN: %target-run %t/ExternalMetadataBuilderTest
+// REQUIRES: executable_test
+// REQUIRES: OS=macosx && CPU=arm64
+
+import ExternalGenericMetadataBuilder
+import Foundation
+import StdlibUnittest
+
+let ExternalGenericMetadataBuilderTests = TestSuite("ExternalGenericMetadataBuilder")
+
+ExternalGenericMetadataBuilderTests.test("JSON output") {
+  let builder = swift_externalMetadataBuilder_create(1, "arm64")
+
+  let inputJSON = """
+  {
+    "metadataNames": [
+      {
+        "name" : "SaySaySdGG"
+      }
+    ]
+  }
+  """
+
+  let readJSONErrorCStr = swift_externalMetadataBuilder_readNamesJSON(builder, inputJSON);
+  let readJSONError = readJSONErrorCStr.map { String(cString: $0) }
+  expectNil(readJSONError)
+
+  // Find libswiftCore dynamically since we don't know where we might have loaded it from.
```

Comment:
`swift_getRuntimeLibraryPath()` will tell you this, FWIW.

Reply:
Thanks, that's much nicer.